pennylane/spin/lattice.py

```diff
@@ -71,12 +71,84 @@ class Lattice:
-    >>> boundary_condition = [True, False]
-    >>> lattice = qml.spin.Lattice(n_cells, vectors, boundary_condition=boundary_condition)
+    We can define the positions of nodes in the lattice unit cell along with the lattice vectors
+    to create a custom lattice layout.
+
+    .. code-block:: python
+
+        from pennylane.spin import Lattice
+
+        positions = [[0.2, 0.5],
+                     [0.5, 0.2],
+                     [0.5, 0.8],
+                     [0.8, 0.5]]
+
+        vectors = [[1, 0], [0, 1]]
+
+        n_cells = [2, 2]
+
+        # periodic boundadry conditions apply along the [1,0] axis only
+        boundary_condition = [True, False]
+
+        lattice = Lattice(n_cells, vectors, positions, boundary_condition=boundary_condition)
+
     >>> lattice.edges
-    [(2, 3, 0), (0, 2, 0), (1, 3, 0), (0, 1, 0)]
+    [(10, 13, 0), (0, 11, 0), (4, 15, 0), (2, 5, 0), (3, 8, 0), (7, 12, 0)]
+
+    .. details::
+        :title: Usage Details
+
+        Unless otherwise specified, the edges will be added based on the ``neighbour_order``,
+        which defaults to 1. Increasing ``neighbour_order`` will add additional connections
+        in the lattice.
+
+        >>> lattice = Lattice(n_cells,
+        ...                   vectors,
+        ...                   positions,
+        ...                   neighbour_order=2,
+        ...                   boundary_condition=boundary_condition)
+        >>> len(lattice.edges)
+        22
+
+        We can also define edges with custom interactions, as well as adding on-site potentials for the
+        nodes:
+
+        .. code-block:: python
+
+            # defining on-site potential at each node in the unit cell
+            custom_nodes = [[(0), ('X', 0.5)],
+                            [(1), ('X', 0.6)],
+                            [(2), ('X', 0.7)],
+                            [(3), ('X', 0.8)]]
+
+            # defining custom edges (instead of nearest-neigbour connections) and their interactions
+            custom_edges = [[(0, 1), ('XX', 0.5)],
+                            [(0, 2), ('YY', 0.6)],
+                            [(1, 3), ('ZZ', 0.7)],
+                            [(2, 3), ('ZZ', 0.7)]]
+
+        >>> lattice = Lattice(n_cells,
+        ...                   vectors,
+        ...                   positions,
+        ...                   custom_edges=custom_edges,
+        ...                   custom_nodes=custom_nodes)
+        >>> lattice.edges
+        [(0, 1, ('XX', 0.5)),
+        (4, 5, ('XX', 0.5)),
+        (8, 9, ('XX', 0.5)),
+        (12, 13, ('XX', 0.5)),
+        (0, 2, ('YY', 0.6)),
+        (4, 6, ('YY', 0.6)),
+        (8, 10, ('YY', 0.6)),
+        (12, 14, ('YY', 0.6)),
+        (1, 3, ('ZZ', 0.7)),
+        (5, 7, ('ZZ', 0.7)),
+        (9, 11, ('ZZ', 0.7)),
+        (13, 15, ('ZZ', 0.7)),
+        (2, 3, ('ZZ', 0.7)),
+        (6, 7, ('ZZ', 0.7)),
+        (10, 11, ('ZZ', 0.7)),
+        (14, 15, ('ZZ', 0.7))]
```

Comment:
Same thing here :( Looks fine with numpy 1.26, but with numpy 2.0:
```python
[(np.int64(0), np.int64(1), ('XX', 0.5)),
 (np.int64(4), np.int64(5), ('XX', 0.5)),
 (np.int64(8), np.int64(9), ('XX', 0.5)),
 (np.int64(12), np.int64(13), ('XX', 0.5)),
 (np.int64(0), np.int64(2), ('YY', 0.6)),
 (np.int64(4), np.int64(6), ('YY', 0.6)),
 (np.int64(8), np.int64(10), ('YY', 0.6)),
 (np.int64(12), np.int64(14), ('YY', 0.6)),
 (np.int64(1), np.int64(3), ('ZZ', 0.7)),
 (np.int64(5), np.int64(7), ('ZZ', 0.7)),
 (np.int64(9), np.int64(11), ('ZZ', 0.7)),
 (np.int64(13), np.int64(15), ('ZZ', 0.7)),
 (np.int64(2), np.int64(3), ('ZZ', 0.7)),
 (np.int64(6), np.int64(7), ('ZZ', 0.7)),
 (np.int64(10), np.int64(11), ('ZZ', 0.7)),
 (np.int64(14), np.int64(15), ('ZZ', 0.7))]
```

Reply:
discussed offline and decided to leave the docstring showing the simpler output